ndarray/src/test/java/org/tensorflow/ndarray/IndexTest.java

```diff
@@ -43,6 +44,371 @@ public void testNullConversions(){
     assertTrue(Indices.slice(null, null).endMask(),
         "Passed null for slice end but didn't set end mask");
   }
+  
+  @Test
+  public void testIndices(){
+
+    String[][] indexData = new String[5][4];
+    for (int i=0 ; i < 5; i++){
+      for (int j=0 ; j < 4; j++)
+        indexData[i][j] = "("+j+", "+i+")";
+    }
+
+    NdArray<String> matrix2d = StdArrays.ndCopyOf(indexData);
+    assertEquals(2, matrix2d.rank());
+
+    /*
+    |(0, 0), (1, 0), (2, 0), (3, 0)|
+    |(0, 1), (1, 1), (2, 1), (3, 1)|
+    |(0, 2), (1, 2), (2, 2), (3, 2)|
+    |(0, 3), (1, 3), (2, 3), (3, 3)|
+    |(0, 4), (1, 4), (2, 4), (3, 4)|  
+    */
+    assertArrayEquals(new String[]{"(0, 0)", "(1, 0)", "(2, 0)", "(3, 0)"}, indexData[0]);
```

Comment:
This line tests that the test data was generated correctly?

Reply:
Yes. Just used the first row of the matrix. Excessive? Should I remove it?

Reply:
No, it's fine. I wanted to check that it was intentional.